mips: ralink: add missing include

namiltd opened the pull request:
Missing headers causes an error on kernel 6.12:
```
arch/mips/ralink/irq.c:86:5: error: no previous prototype for 'get_c0_perfcount_int' [-Werror=missing-prototypes]
   86 | int get_c0_perfcount_int(void)
      |     ^~~~~~~~~~~~~~~~~~~~
arch/mips/ralink/irq.c:92:14: error: no previous prototype for 'get_c0_compare_int' [-Werror=missing-prototypes]
   92 | unsigned int get_c0_compare_int(void)
      |              ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

namiltd commented:
Detected and created during: https://github.com/openwrt/openwrt/pull/18654

KanjiMonster commented on target/linux/generic/hack-6.12/931-mips-ralink-add-missing-include.patch:
```diff
@@ -0,0 +1,17 @@
+From b71ab42067204c3d90aa9143b815482b222fee32 Mon Sep 17 00:00:00 2001
```
Why is this a "hack"? If this issue is introduced by one of our patches, the fix should go in there, if it is upstream, it should be sent upstream. And either way it isn't a hack.

namiltd replied:
If not put the fix in the hack then where?

namiltd replied:
So please send this patch to upstream.

tmn505 replied:
As @KanjiMonster said, You are putting this in a wrong place, it should be added in `target/linux/ramips/patches-*/311-MIPS-use-set_mode-to-enable-disable-the-cevt-r4k-irq.patch`.

namiltd replied:
Ok I see.

KanjiMonster commented on target/linux/generic/hack-6.12/931-mips-ralink-add-missing-include.patch:
```diff
@@ -0,0 +1,17 @@
+From: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+Date: Mon, 18 Nov 2024 21:28:05 +0100
+Subject: mips: ralink: add missing include
+
```
Please also add the commit message here, so it is part of the patch.

namiltd replied:
Done

namiltd committed:
mips: ralink: add missing include

Missing headers causes an error on kernel 6.12:
arch/mips/ralink/irq.c:86:5: error: no previous prototype for 'get_c0_perfcount_int' [-Werror=missing-prototypes]
   86 | int get_c0_perfcount_int(void)
      |     ^~~~~~~~~~~~~~~~~~~~
arch/mips/ralink/irq.c:92:14: error: no previous prototype for 'get_c0_compare_int' [-Werror=missing-prototypes]
   92 | unsigned int get_c0_compare_int(void)
      |              ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
Moved to PR https://github.com/openwrt/openwrt/pull/18654
